Signal/src/ViewControllers/BlockListViewController.m

```diff
@@ -130,6 +134,24 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
     return cell;
 }
 
+- (NSString *)displayNameForIndexPath:(NSIndexPath *)indexPath
+{
+    NSString *phoneNumber = _blockedPhoneNumbers[(NSUInteger)indexPath.item];
+    PhoneNumber *parsedPhoneNumber = [PhoneNumber tryParsePhoneNumberFromUserSpecifiedText:phoneNumber];
+
+    // Try to parse and present the phone number in E164.
+    // It should already be in E164, so this should always work.
+    // If an invalid or unparsable phone number is already in the block list,
+    // present it as-is.
+    NSString *displayName = (parsedPhoneNumber ? parsedPhoneNumber.toE164 : phoneNumber);
```

Comment:
nit: Consider duduplicating some of this logic by using `[contactManager displayNameForPhoneIdentifier:]` when the number is parseable as e164.